rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -83,520 +90,390 @@ fn oversized_reject_message_is_truncated() {
     );
 }
 
+/// Tests that inducting a loopback stream where `signals_end` < `end` panics (this must be equal
+/// for the loopback stream, because it is its own reverse stream).
 #[test]
 #[should_panic(
     expected = "Invalid message indices in stream slice from subnet g24bn-xymaa-aaaaa-aaaap-yai: messages begin (21) != stream signals_end (22)"
 )]
-fn induct_loopback_stream_with_signals_panics() {
-    with_test_replica_logger(|log| {
-        let (stream_handler, mut state, _) = new_fixture(&log);
-
-        // A loopback stream containing 2 messages and 1 signal.
-        let loopback_stream = generate_loopback_stream(StreamConfig {
-            messages_begin: 21,
-            message_count: 2,
+fn induct_loopback_stream_with_signals_end_before_messages_end_panics() {
+    let (stream_handler, state, _) =
+        with_local_test_setup(btreemap![LOCAL_SUBNET => StreamConfigX {
+            begin: 21,
+            messages: vec![*LOOPBACK_REQUEST_CONFIG; 2],
```

Comment:
Nit: I'd rather spell out `[*LOOPBACK_REQUEST_CONFIG, *LOOPBACK_REQUEST_CONFIG]`, it takes a lot less effort to eyeball than `[*LOOPBACK_REQUEST_CONFIG; 2]`.

Also, looking at it in this context, whether the struct is called `MessageConfigX` or not, it makes a lot more sense for the constant / static variable to be called `LOOPBACK_REQUEST` than `LOOPBACK_REQUEST_CONFIG`. (Or you can just replace it with something like `request(*LOCAL_CANISTER, *LOCAL_CANISTER)`, as suggested elsewhere. Up to you.)

Reply:
I'm not using the `vec![a; 2]` syntax anymore though I don't see what the problem is TBH.

`MessageConfigX` is now called `MessageBuilder` since it's basically a wrapper around `RequestBuilder`, `ResponseBuilder`.

Is is then later on used in `StreamConfig` like so:
```
StreamConfig {
   messages: vec![
      BuildRequest(*LOCAL_CANISTER, *LOCAL_CANISTER),
      BuildResponse(*LOCAL_CANISTER, *OTHER_LOCAL_CANISTER),
   ],
   ..StreamConfig::default()
}
```
Which is supposed to mean: 'when you make a stream for me please build a request from here to there and build a response from here to there`...

I also ditched the constants for `LOOPBACK_REQUEST` and so on.